### PR TITLE
[BE]: Update flake8-comprehensions and enable C420

### PR DESCRIPTION
**Pull request by Skylion007**
Uses `dict.fromkeys` whenever possible as covered by flake8-comprehensions rule C420. While the ruff rule RUF025 is still in preview, flake8-comprehensions have added a new rule which covers this. Use dict.fromkeys is faster when the value being added to the dictionary is the same at every iteration and is immutable, it also removes an unnecessary dict comprehension.

This rule will be enabled with our current ruleset in RUF in 0.6 as C420.

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: test/jit/test_list_dict.py [hunk 1]
```diff
@@ -1645,9 +1645,9 @@ def test_dictliteral_is_typed_from_annotation():
 
         def test_dictcomprehension_is_typed_from_annotation():
             metasyntactics = ["foo", "bar", "baz"]
-            x: Dict[str, Optional[int]] = {
+            x: Dict[str, Optional[int]] = {  # noqa: C420, RUF025
                 word: None for word in metasyntactics
-            }  # noqa: RUF025
+            }
```
Comment (lezcano) [inline test/jit/test_list_dict.py:1650]: why not
```python
  x: Dict[str, Optional[int]] = dict.fronkeys(metasyntactics)
```

Comment (Skylion007) [reply test/jit/test_list_dict.py:1650]: @lezcano Because it's testing JIT features and I didn't want to touch it.

**Review by lezcano — APPROVED** [review]

**Review by ezyang — APPROVED** [review]

**Pull request closed without merging**